### PR TITLE
Add swagger

### PR DESCRIPTION
**Pull request by confleux**

### CODE REVIEW

**Comment by bludnic** [comment]
`type` prop of `TokenAsset` can be described as enum: "coin" | "ERC20"

File: specification/openapi.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    "version": "1.0.0",
+    "title": "Swagger Adamant-wallets"
+  },
+  "paths": {},
+  "components": {
+    "schemas": {
+      "TokenGeneral": {
+        "allOf": [
+          {
+            "$ref": "#/components/schemas/TokenGeneral"
```
Comment (bludnic) [inline specification/openapi.json:13]: Doesn't make sense to ref the component to itself. Remove `$ref` and `allOf`.

File: specification/openapi.json [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                }
+              }  
+            }
+          }
+        ]
+      },
+      "TokenAsset": {
+        "allOf": [
+          {
+            "$ref": "#/components/schemas/TokenAsset"
```
Comment (bludnic) [inline specification/openapi.json:98]: remove `$ref` and `allOf`

File: specification/openapi.json [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                "enum": [
+                  "coin",
+                  "ERC20"
+                ]
+              },    
+              "decimals": {
+                "type": "integer"
+              },
+              "minBalance": {
+                "type": "integer"
```
Comment (bludnic) [inline specification/openapi.json:56]: should be type `number` because it's a floating-point number

File: specification/openapi.json [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+              },
+              "cryptoTransferDecimals": {
+                "type": "integer"
+              },
+              "minTransferAmount": {
+                "type": "integer"
```
Comment (bludnic) [inline specification/openapi.json:62]: `number`

File: specification/openapi.json [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+              },
+              "fixedFee": {
+                "type": "integer"
```
Comment (bludnic) [inline specification/openapi.json:65]: `number`

File: specification/openapi.json [hunk 6]
```diff
@@ -0,0 +1,147 @@
+{
+  "openapi": "3.0.0",
```
Comment (bludnic) [inline specification/openapi.json:2]: Latest version is `3.0.3` according to https://swagger.io/specification/

File: specification/openapi.json [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                "type": "integer"
+              },
+              "minTransferAmount": {
+                "type": "integer"
+              },
+              "fixedFee": {
+                "type": "integer"
+              },  
+              "qqPrefix": {
+                "type": "string"
```
Comment (bludnic) [inline specification/openapi.json:68]: What is this prop for? It's not clear for me. Add a description maybe.

```suggestion
                "type": "string",
                "description": "Property description"
```

**Review by bludnic — CHANGES_REQUESTED** [review]

**Review by bludnic — APPROVED** [review]

**Review by adamant-al — CHANGES_REQUESTED** [review]
Update Swagger scheme

**Review by martiliones — APPROVED** [review]

**Review by adamant-al — APPROVED** [review]

**Comment by adamant-al** [comment]
@confleux Resolve conflicts.
@bludnic  Review.

File: specification/openapi.json [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+          },
+          "txConsistencyMaxTime": {
+            "type": "integer",
+            "description": "Time in ms when difference between in-chat transfer and Tx timestamp considered as acceptable"
+          },
+          "nodes": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
```
Comment (bludnic) [inline specification/openapi.json:163]: 1. Add [alt_ip](https://github.com/Adamant-im/adamant-wallets/blob/master/assets/general/adamant/info.json#L27) and [hasIndex](https://github.com/Adamant-im/adamant-wallets/blob/master/assets/general/ethereum/info.json#L37). Both are optional
2. Set `url` as required param

File: specification/openapi.json [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+                  "type": "string"
+                }
+              }
+            },
+            "description": "Node links for API"
+          },
+          "links": {
+            "type": "array",
+            "items": {
+              "type": "object",
```
Comment (bludnic) [inline specification/openapi.json:174]: The array is optional but the object properties must be required. At least the `url` param.

```suggestion
              "type": "object",
              "required": ["name", "url"],
```

**Review by bludnic — CHANGES_REQUESTED** [review]

File: specification/openapi.json (hunk 8 above)
Comment (bludnic) [reply specification/openapi.json:163]: @martiliones what is `hasIndex` for? There is no info in the README.md

Comment (confleux) [reply specification/openapi.json:166]: Probably, indicates whether [ETH Indexer](https://github.com/Adamant-im/ETH-transactions-storage) installed

**Review by bludnic — APPROVED** [review]

**Pull request merged**